website/docs/pact_nirvana/step_7.md

```diff
@@ -22,36 +31,78 @@ Another reason that an extra "safe to deploy?" check is required (as well as the
-3. Add a step to your deployment process so that when the application is deployed to production, the relevant pacticipant version in the broker is tagged as the “production” version. This functionality is provided by the [pact-broker client CLI](/pact_broker/client_cli/readme#create-version-tag)
 
-Useful links:
+## Lets tag on deploy!
 
-* [Checking verification results using the can-i-deploy CLI](/pact_broker/can_i_deploy/)
-* [can-i-deploy CLI documentation](/pact_broker/client_cli/readme#can-i-deploy)
+Now we're ready to tag pacts when we deploy, and switch `can-i-deploy` to use environment names rather than branch names.
+
+<details open>
+  <summary>Diamond level diagram</summary>
+
+![Diamond level diagram](images/diamond.png)
+</details>
+
+Note how in the PR pipeline we are checking to see if we can deploy to all the environments that we deploy to from 
+the main branch. The rationale for this is that we don't want to merge something into our main branch which we know 
+will break once we deploy - we want our main branch to stay clean and not get blocked by broken builds.  If you 
+deploy to other environments such as preprod, you would want to check that environment as well.
+
+Then, when we're actually ready to deploy to an environment, we check again. We do this because the version of the 
+other side of the contract could have changed since the PR validation ran.  Doing the check in the PR pipeline 
+reduces the chance of breakage, but it's still a possibility, and we want to check for that before we deploy.
+
+Note also that the first step in our commit pipeline is to let the broker know that a particular version of the 
+provider or consumer is now in the main branch.
+
+### Add tagging to your commit pipelines
 
+The first step is to start tagging when the consumer or provider is deployed to an environment. So when you successfully 
+deploy to staging, apply the staging tag; when you successfully deploy to prod, apply the prod tag. Do this for each 
+environment you deploy to. 
 
-## Setting up a Pact in a "universal" CI/CD pipeline
+1. Add a step to your deployment process so that when the consumer is deployed to an environment, the relevant pacticipant version in the broker is tagged with the name of the environment (eg. `test`, `production`). This functionality is provided by the [pact broker CLI](/pact_broker/client_cli/readme#create-version-tag) which is available as a standalone executable, and a Docker image.
+2. Add the new tag names to the list of pacts the provider will verify (eg. `{ consumerVersionSelectors: { tag: "main", latest: true }, { tag: "test", latest: true }, { tag: "production", latest: true } }`).
```

Comment:
I don't understand this step.

Reply:
So this comes from [step 6](https://docs.pact.io/pact_nirvana/step_6) in the current road to pact nirvana.

It includes the first mention to consumer version selectors

https://docs.pact.io/pact_broker/advanced_topics/consumer_version_selectors

> Consumer version selectors are the (new) way to configure which pacts the provider verifies. Instead of providing a list of tag names (as in the past) a list of selector objects is provided. These allow a much more flexible and powerful approach to specifying which pacts to verify.

It's worth a link to those docs there for sure 👍🏾

Reply:
Oh, wow, I was completely unaware of this! I need to bone up on this...

Reply:
Just having a quick read of this, I don't think step 2 belongs here. The mention of the universal pipeline is for when you're all forced to use the same pipeline, so can-i-deploy runs every time whether the app actually uses pact or not. Step 1 is there so that the broker knows about the app version in that env, so the can-i-deploy doesn't blow up. There's nowhere to put the selectors in that case.

Reply:
Sorry, ignore that comment, the diff was confusing me.